### PR TITLE
Switch builds to BuildBox

### PR DESCRIPTION
**Pull request by xbrianh**
- Add triggers to `.travis.yml` to initiate builds on the Build Box

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -78,6 +81,11 @@ jobs:
       skip_cleanup: true
       on:
         all_branches: true  # but see the `if:` conditional on the deploy stage above
+  - stage: deploy-dev
+    script:
+    - "http POST ${GITLAB_API}/projects/${REPO_SLUG}/mirror/pull Private-Token:${GITLAB_TOKEN}"
+    - sleep 60  # give the build box some breathing room to update repo
```
Comment (hannes-ucsc) [inline .travis.yml]: Is there any way we can use some sort of polling here? Or trigger the pipeline automatically by the poll?

Comment (hannes-ucsc) [reply .travis.yml]: … automatically by the ~poll~ pull.

Comment (xbrianh) [reply .travis.yml]: We’ve explored build scheduling through GitLab, and of course there are other ways. Triggering from Travis looks to be the simplest by far.

Comment (hannes-ucsc) [reply .travis.yml]: When you trigger a pull via HTTP POST, shouldn't the new commits automatically trigger a build?

File: scripts/trigger_gitlab_build.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+repo=$2
+branch=$3
+slug=$(python -c "import sys, urllib.parse; print(urllib.parse.quote_plus(sys.argv[1]))" "${owner}/${repo}")
+
+for i in $(seq 3); do
+	GITHUB_API=https://api.github.com
+	GH_COMMITS=$(http GET ${GITHUB_API}/repos/${owner}/${repo}/commits sha==${branch})
+	GH_LATEST=$(echo ${GH_COMMITS} | jq -r '.[0]["sha"]')
+	
+	GL_COMMITS=$(http GET ${GITLAB_API}/projects/${slug}/repository/commits ref_name==${branch} PRIVATE-TOKEN:${GITLAB_TOKEN})
```
Comment (kislyuk) [inline scripts/trigger_gitlab_build.sh]: Where is `$GITLAB_API` set?

File: scripts/trigger_gitlab_build.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -euo pipefail
+if [[ $# != 3 ]]; then
+	echo "This script triggers allspark build for owner, repo, and branch"
```
Comment (hannes-ucsc) [inline scripts/trigger_gitlab_build.sh]: One indent too many.

**Pull request closed without merging**